src/Ledger/Utxo.lagda

```diff
@@ -130,23 +130,24 @@ instance
   HasCoin-UTxO : HasCoin UTxO
   HasCoin-UTxO .getCoin = cbalance
 
+-- module _ (let open TxBody) where
 \end{code}
 \begin{code}
+certDeposit : DCert → {pp : PParams} → DepositPurpose ⇀ Coin
+certDeposit (delegate c _ _ v)  = ❴ CredentialDeposit c , v                ❵
+certDeposit (regpool c _) {pp}  = ❴ PoolDeposit       c , pp .poolDeposit  ❵
+certDeposit (regdrep c v _)     = ❴ DRepDeposit       c , v                ❵
+certDeposit _                   = ∅
+
+certRefund : DCert → ℙ DepositPurpose
+certRefund (dereg c)      = ❴ CredentialDeposit c ❵
+certRefund (deregdrep c)  = ❴ DRepDeposit c ❵
+certRefund _              = ∅
```

Comment:
Exposing `certDeposit` and `certRefund` to use them in proof of ga deposits invariant.